File: examples/traits/tests/bindings/test_traits.py
from traits import *

for button in get_buttons():
    name = button.name()
    # Check that the name is one of the expected values
    assert(name in ["go", "stop"])
    # Check that we can round-trip the button through Rust
    assert(press(button).name() == name)

# Test a button implemented in Python
class PyButton(Button):
    def name(self):
        return "PyButton"

assert(press(PyButton()).name() == "PyButton")

File: examples/traits/tests/bindings/test_traits.swift
import traits

for button in getButtons() {
    let name = button.name()
    // Check that the name is one of the expected values
    assert(["go", "stop"].contains(name))
    // Check that we can round-trip the button through Rust
    assert(press(button: button).name() == name)
}

// Test a Button implemented in Swift
final class SwiftButton: Button {
    func name() -> String {
        return "SwiftButton"
    }
}

assert(press(button: SwiftButton()).name() == "SwiftButton")
